x/ibc/types.go

```diff
@@ -0,0 +1,110 @@
+package ibc
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	wire "github.com/cosmos/cosmos-sdk/wire"
+)
+
+// ------------------------------
+// IBCPacket
+
+// IBCPacket defines a piece of data that can be send between two separate
+// blockchains.
+type IBCPacket struct {
+	SrcAddr   sdk.Address
+	DestAddr  sdk.Address
+	Coins     sdk.Coins
+	SrcChain  string
+	DestChain string
+}
+
+func NewIBCPacket(srcAddr sdk.Address, destAddr sdk.Address, coins sdk.Coins,
+	srcChain string, destChain string) IBCPacket {
+
+	return IBCPacket{
+		SrcAddr:   srcAddr,
+		DestAddr:  destAddr,
+		Coins:     coins,
+		SrcChain:  srcChain,
+		DestChain: destChain,
+	}
+}
+
+func (ibcp IBCPacket) ValidateBasic() sdk.Error {
+	if ibcp.SrcChain == ibcp.DestChain {
+		return ErrIdenticalChains().Trace("")
+	}
+	return nil
+}
+
+// ----------------------------------
+// IBCTransferMsg
+
+// IBCTransferMsg defines how another module can send an IBCPacket.
+type IBCTransferMsg struct {
+	IBCPacket
+}
+
+func (msg IBCTransferMsg) Type() string {
+	return "ibc"
+}
+
+func (msg IBCTransferMsg) Get(key interface{}) interface{} {
+	return nil
+}
+
+func (msg IBCTransferMsg) GetSignBytes() []byte {
+	cdc := wire.NewCodec()
+	bz, err := cdc.MarshalBinary(msg.IBCPacket)
```

Comment:
I think we'd want to marshal the entire msg, not just the packet!

In any case it's fine because we can get rid of GetSignBytes soon and just put the entire Msg in the StdSignDoc